src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -652,11 +659,15 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
     withWorkspaceContextAndPermissions(methodRepoQuery.source.workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
       withMethodConfig(workspaceContext, methodRepoQuery.source.namespace, methodRepoQuery.source.name, txn) { methodConfig =>
         import org.broadinstitute.dsde.rawls.model.MethodRepoJsonSupport._
-        RequestComplete(methodRepoDAO.postMethodConfig(
+        methodRepoDAO(context).postMethodConfig(
           methodRepoQuery.methodRepoNamespace,
           methodRepoQuery.methodRepoName,
           methodConfig.copy(namespace = methodRepoQuery.methodRepoNamespace, name = methodRepoQuery.methodRepoName),
-          userInfo))
+          userInfo) {
+
+          result => context.parent ! RequestComplete(result.get)
+        }
+        RequestComplete("done")
```

Comment:
haven't figured out what to do with it yet

On Tuesday, September 29, 2015, jmthibault79 <notifications@github.com>
wrote:

> In
> src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala
> <https://github.com/broadinstitute/rawls/pull/148#discussion_r40698371>:
>
> >            methodRepoQuery.methodRepoNamespace,
> >            methodRepoQuery.methodRepoName,
> >            methodConfig.copy(namespace = methodRepoQuery.methodRepoNamespace, name = methodRepoQuery.methodRepoName),
> > -          userInfo))
> > +          userInfo) {
> > +
> > +          result => context.parent ! RequestComplete(result.get)
> > +        }
> > +        RequestComplete("done")
>
> Does this line serve a purpose other than satisfying the return type?
> Would we still use PerRequestMessage if we moved over to this system?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/rawls/pull/148/files#r40698371>.
>